corehq/apps/hqwebapp/signals.py

```diff
@@ -0,0 +1,26 @@
+from datetime import date
+
+from django.contrib.auth.signals import user_logged_in, user_login_failed
+from django.dispatch import receiver
+
+from corehq.apps.users.models import CouchUser
+
+@receiver(user_logged_in)
+def clear_failed_logins_and_unlock_account(sender, request, user, **kwargs):
+    user = CouchUser.from_django_user(user)
+    if user.is_web_user():
```

Comment:
I think you could add `and user.login_attempts > 0` here so that you don't need to save the user if they don't have failed logins.